pkg/engine/vmextensions_test.go

```diff
@@ -222,6 +222,7 @@ func TestCreateCustomScriptExtension(t *testing.T) {
 				BlockOutboundInternet:    false,
 				EnableCSERunInBackground: false,
 			},
+			MasterProfile: &api.MasterProfile{},
```

Comment:
curious, what's the purpose of this addition?

Reply:
Was to fix a null pointer ref in one of the tests, although now that I'm writing that out I want to check again what happens if I remove this...

Reply:
makes sense, thanks for confirming!

Reply:
Removing this, I think it was because in VMSS I didn't account for HostedMasters, which I since changed.